javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AutoResourceProvider.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.tooling;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.autoconfigure.spi.ResourceProvider;
+import io.opentelemetry.sdk.resources.Resource;
+
+@AutoService(ResourceProvider.class)
+public class AutoResourceProvider implements ResourceProvider {
+
+  private static final String TELEMETRY_AUTO_VERSION = "telemetry.auto.version";
```

Comment:
Take the propery names from `import static io.opentelemetry.semconv.resource.attributes.ResourceAttributes`